### PR TITLE
feat: add conversation id on error messages in amazonqFeatureDev

### PR DESCRIPTION
**Pull request by sannicm**
## Problem

Even with the conversation id in the info logs, it could be cumbersome for a customer to get it. From UX feedback it was proposed to add the conversation on the error messages to be more specific and direct, and make the sharing of the conversation Id to customer support more convenient.

## Solution

if the session has a conversation id, error messages will show the id with an accompanying text.

How the message will look:

Screenshot 2024-05-15 at 14 35 26

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinmk3 — APPROVED** [review]
LGTM after feature dev team approval

File: packages/core/src/amazonqFeatureDev/controllers/chat/messenger/messenger.ts
Comment (sannicm) [reply packages/core/src/amazonqFeatureDev/controllers/chat/messenger/messenger.ts:66]: I've had some feedback from tech writer and UX, the message will be updated

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts
Comment (sarayev) [inline packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:197]: non-blocker: I wonder if session can be obtained before this line and the conversationId would be incliuded in this error, too. 

Something we can discuss offline, not a big deal atm.

**Review by sarayev — APPROVED** [review]

**Pull request merged**
